Add bulk methods section (#145)
AlexsLemonade/ScPCA-manuscript: content/04.methods.md

If a merged `SingleCellExperiment` object contained any ADT data, the RNA and ADT data were exported and saved separately as RNA (`_rna.h5ad`) and ADT (`_adt.h5ad`) objects.
In contrast, if a merged `SingleCellExperiment` object contained HTO data due to the presence of any multiplexed libraries in the merged object, the HTO data was removed from the `SingleCellExperiment` object and not included in the exported `AnnData` object.

### Analysis of bulk RNA-seq data

#### Data preparation

We identified solid tumor samples with both bulk and single-cell (or single-nuclei) RNA-seq data in the ScPCA Portal for analysis, with multiplexed samples excluded (N=105).

> We further identified low-quality samples based on visual inspection of quality control reports and removed (N=8), leaving a total of 97 samples across five ScPCA projects for analysis.

Reviewer: ```suggestion
We removed low-quality samples based on visual inspection of quality control reports (N=8), leaving a total of 97 samples across five ScPCA projects for analysis.
```

> For each project, we transformed and normalized with respect to library size bulk counts matrices for all samples using `DESeq2::rlog()` [@doi:10.1186/s13059-014-0550-8].

Reviewer: I don't know it this goes too far on simplification, but the "with respect to library size" phrase just felt really awkward. And I think perhaps saying that we used `rlog` is sufficient to say what we were doing?
```suggestion
For each project, we transformed and normalized bulk counts matrices for all samples using `DESeq2::rlog()` [@doi:10.1186/s13059-014-0550-8].
```

Author: I don't think this goes too far, especially since we have open code for this

We obtained pseudobulk counts by summing raw single-cell counts for each sample, and similarly transformed each project's resulting counts matrix with `DESeq2::rlog()`.
We filtered out genes which were not observed in either the bulk or pseudobulk raw counts matrices before subsequent analysis.
For each project, we then used the `lme4` [@doi:10.18637/jss.v067.i01] R package to construct a linear model predicting bulk from pseudobulk counts considering a random effect for sample id: `bulk ~ pseudobulk + (1|sample_id)`.

#### Overrepresentation analysis

> To identify cell types that might be overrrepresented in one modality compared to the other, we first identified gene sets associated with each cell type of interest, defined as each project's consensus cell types.

Reviewer: I think we want to separate the selection of the cell types we are looking at for a project from the gene sets that are tested.  So first we select cell types of interest using conensus cell types for the project. Then we identify the gene set for each cell type. I think that will clarify?

(my suggestion below is only about removing the extra "r")
```suggestion
To identify cell types that might be overrepresented in one modality compared to the other, we first identified gene sets associated with each cell type of interest, defined as each project's consensus cell types.
```

Author: > (my suggestion below is only about removing the extra "r")

I have two very mediocre, not fully formed jokes:
- so many r's this could be an r package!
- yes of course, the r's are overrepresented

Because the consensus cell types can encompass multiple cell types from the `CellAssign` marker gene reference, we defined the consensus cell type gene sets as the union of all gene sets for the constituent cell types.

> For input to the overrepresentation analysis, we summarized model residuals by taking the median residual for each gene across samples and then transforming these summarized residuals into Z-scores.

Reviewer: ```suggestion
For input to the overrepresentation analysis, we summarized model residuals by taking the median residual for each gene across samples within each project and then transformed these summarized residuals into Z-scores.
```

We identified outlier genes as those with Z-scores greater than 2.5 (positive outliers) or less than -2.5 (negative outliers).
In this case, positive outliers represent genes with comparatively higher expression in the bulk modality, and negative outliers represent genes with comparatively higher expression in the single-cell modality.

For each consensus cell type gene set, we calculated two odds ratios representing whether genes were overrepresented in the positive outliers (enriched in bulk) or negative outliers (enriched in pseudobulk).
We calculated P-values for both the bulk and pseudobulk enrichment directions via permutation testing with 10,000 replicates.

> We defined gene sets with significant overrepresentation as those with a false-discovery-rate--corrected P-value ≤ 0.05 [@doi:10.1111/j.2517-6161.1995.tb02031.x].

Reviewer: ```suggestion
We defined gene sets with significant overrepresentation as those with a false-discovery-rate-corrected P-value ≤ 0.05 [@doi:10.1111/j.2517-6161.1995.tb02031.x].
```

Low-quality samples (N=8) based on visual inspection of quality control reports were then removed, leaving a total of 97 samples across five ScPCA projects for analysis.
All analyses described here were conducted separately for each project.

We obtained the bulk raw counts matrix for all samples of interest from Portal and transformed them with `DESeq2::rlog()` [@doi:10.1186/s13059-014-0550-8].
In addition, we established a matrix of pseudobulk counts by summing all raw single-cell counts in relevant processed objects from the ScPCA Portal, and we similarly transformed the resulting counts matrix with `DESeq2::rlog()`.
Genes which were not observed in either the bulk or pseudobulk raw counts matrices were filtered out before subsequent analysis.
For each project, we then used the `lme4` [@doi:10.18637/jss.v067.i01] R package to construct a linear model predicting bulk from pseudobulk counts considering a random effect for sample id: `bulk ~ pseudobulk + (1|sample_id)`.

#### Overrepresentation analysis

We next performed an overrepresentation analysis on these model's residuals to test whether certain cell types were overrepresented in one modality compared to the other.

> To this end, we summarized model residuals into per-gene quantities by taking the median residual across samples and then transforming these summarized residuals into Z-scores.

Reviewer: ```suggestion
To this end, we summarized model residuals within each project by taking the median residual for each gene across samples, then transforming these summarized residuals into Z-scores.
```

> From each project's distribution of summarized residuals, we identified positive and negative outliers as those values greater than 2.5 or less than -2.5, respectively.

Reviewer: ```suggestion
We identified outlier genes as those with Z-scores greater than 2.5 (positive outliers) or less than -2.5 (negative outliers).
```

#### Data preparation

We identified solid tumor samples with both bulk and single-cell (or single-nuclei) RNA-seq data in the ScPCA Portal for analysis, with multiplexed samples excluded (N=105).
We removed low-quality samples based on visual inspection of quality control reports (N=8), leaving a total of 97 samples across five ScPCA projects for analysis.

For each project, we transformed and normalized bulk counts matrices for all samples using `DESeq2::rlog()` [@doi:10.1186/s13059-014-0550-8].
We obtained pseudobulk counts by summing raw single-cell counts for each sample, and similarly transformed each project's resulting counts matrix with `DESeq2::rlog()`.
We filtered out genes which were not observed in either the bulk or pseudobulk raw counts matrices before subsequent analysis.
For each project, we then used the `lme4` [@doi:10.18637/jss.v067.i01] R package to construct a linear model predicting bulk from pseudobulk counts considering a random effect for sample id: `bulk ~ pseudobulk + (1|sample_id)`.

#### Overrepresentation analysis

We next asked whether certain cell types might be overrepresented in one modality compared to the other.

> For this, we first identified cell types of interest as all possible consensus cell types for a given project.

Reviewer: I'm not sure that this is accurate, but I think we want to say something mor explict than "for a given project"
```suggestion
For this, we first identified cell types of interest as all possible consensus cell types that were labeled in each project.
```

Author: Yeah, this isn't quite accurate - we did include cell types that were not labeled, but _could_ have been labeled, but I worry now that "could have been labeled" may just raise more questions than it answers. Maybe I'll add a TODO here to really remind us to check phrasing.

> We then established gene sets for each consensus cell type as the union from marker genes in the project's `CellAssign` marker gene reference.

Reviewer: The next sentence covers the union, so I think this can be shortened
```suggestion
We then created gene sets for each consensus cell type using the project's `CellAssign` marker gene reference.
```

Because a consensus cell type can encompass multiple cell types in the marker gene reference, we defined each consensus cell type's gene set as the union of all marker genes for each of its constituent reference cell types.

> For input to the overrepresentation analysis, we summarized model residuals by taking the median residual for each gene across samples within each project and then transformed these summarized residuals into Z-scores.

Reviewer: slight rearrangement
```suggestion
For input to the overrepresentation analysis, we summarized model residuals within each project by taking the median residual for each gene across samples and then transformed these summarized residuals into Z-scores.
```
